### PR TITLE
Decompiled various functions

### PR DESCRIPTION
**Pull request by Brotenko**
Decompiled the following functions in `1AC760`:
- dispatch_event_partner
- dispatch_event_partner_continue_turn
- func_8028070C

### CODE REVIEW

File: include/common_structs.h [hunk 1]
```diff
@@ -1723,4 +1723,15 @@ typedef struct {
     /* 0x11630 */ Matrix4s matrixStack[0x200];
 } DisplayContext; // size = 0x19630
 
+typedef struct EffectInstanceData {
+    /* 0x00 */ s32 unk_00;
+    /* 0x0C */ Vec3f pos;
```
Comment (ethteck) [inline include/common_structs.h:1728]: pos has the offset 0x4, not 0xC

File: src/code_1AC760.c [hunk 2]
```diff
@@ -1,23 +1,124 @@
 #include "common.h"
 #include "battle/battle.h"
 
-INCLUDE_ASM(s32, "code_1AC760", dispatch_event_partner);
+void dispatch_event_partner(s8 lastEventType) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    s32 temp_ret;
```
Comment (ethteck) [inline src/code_1AC760.c:6]: temp_ret can be removed

File: src/code_1AC760.c [hunk 3]
```diff
@@ -42,10 +143,49 @@ INCLUDE_ASM(s32, "code_1AC760", PartnerPowerBounceEnemy);
...
+    Bytecode* args = script->ptrReadPos;
+    Actor* actor = get_actor(script->owner1.actorID);
+    s32 damageAmount = get_variable(script, *args++);
+    s32 event = get_variable(script, *args++);
+    
+    battleStatus->currentTargetID = actor->targetActorID;
+    battleStatus->currentTargetPart = actor->targetPartIndex;
+
+    if (dispatch_damage_event_partner_0(damageAmount, event, battleStatus) >= 0) {
+        return does_script_exist_by_ref(script) ? ApiStatus_DONE2 : ApiStatus_BLOCK;
```
Comment (ethteck) [inline src/code_1AC760.c:157]: I'd rather use not ternaries if possible

File: src/code_1AC760.c [hunk 4]
```diff
@@ -42,10 +143,49 @@ INCLUDE_ASM(s32, "code_1AC760", PartnerPowerBounceEnemy);
...
+    }
+
+    return ApiStatus_BLOCK;
+}
+
+ApiStatus DeletePartner(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
```
Comment (ethteck) [inline src/code_1AC760.c:164]: Empty line after declaration ples

File: src/code_1AF230.c [hunk 5]
```diff
@@ -1,3 +1,21 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "code_1AF230", LoadStarPowerScript);
+typedef struct StarPower { void* dmaStart; void* dmaEnd; void* dmaDest; Bytecode* init; } StarPower;
```
Comment (ethteck) [inline src/code_1AF230.c:3]: Could you write these in the same style we do other structs (fields on their own line, comments for offsets)?

**Review by ethteck — CHANGES_REQUESTED** [review]
Awesome work! A few things but looks great

File: src/code_1AC760.c (hunk 3 above)
Comment (nanaian) [reply src/code_1AC760.c:157]: I disagree, unless the `if else` version doesn't match.
IIRC this was a classic `(bool != 0) * 2` thing we see fairly often - and we should change the other ones to look like this / `if return else return` if that matches.

Comment (ethteck) [reply src/code_1AC760.c:157]: discussed in chat and I agree that in this case they make sense

File: src/code_1AF230.c (hunk 5 above)
Comment (nanaian) [reply src/code_1AF230.c:3]: Also this should be named something like "StarPowerEntry"

File: src/code_1AF230.c [hunk 6]
```diff
@@ -1,3 +1,21 @@
...
+
+extern StarPower D_8029C7D0[13];
+
+typedef struct WhateverThisIs { s8 starpoints; s8 a; s8 b; s8 c; s32 w; s32 x; s32 y; s32 z; } WhateverThisIs;
```
Comment (nanaian) [inline src/code_1AF230.c:7]: Name this "StarPowerDesc" or similar. The first field should be `starPoints` also

File: src/code_1AF230.c [hunk 7]
```diff
@@ -1,3 +1,21 @@
...
+
+extern WhateverThisIs D_8008F071[];
+
+ApiStatus LoadStarPowerScript(ScriptInstance *script, s32 isInitialCall) {
+    BattleStatus *battleStatus = &gBattleStatus;
+    PlayerData *playerData = &gPlayerData;
+    s16 selectedItemID;
+
+    playerData->specialBarsFilled -= D_8008F071[battleStatus->selectedMoveID].starpoints * 256;
```
Comment (nanaian) [inline src/code_1AF230.c:16]: out-of-scope comment: we should make this * 256 into a #define or something.

**Review by ethteck — DISMISSED** [review]

File: src/code_1AF230.c [hunk 8]
```diff
@@ -1,19 +1,33 @@
...
-typedef struct WhateverThisIs { s8 starpoints; s8 a; s8 b; s8 c; s32 w; s32 x; s32 y; s32 z; } WhateverThisIs;
+typedef struct StarPowerDesc {
+    /* 0x00 */ s8 starPoints;
+    /* 0x01 */ s8 a;
+    /* 0x02 */ s8 b;
+    /* 0x03 */ s8 c;
+    /* 0x04 */ s32 w;
+    /* 0x08 */ s32 x;
+    /* 0x0C */ s32 y;
+    /* 0x10 */ s32 z;
```
Comment (nanaian) [inline src/code_1AF230.c:20]: Maybe call these `unk_xx`? Doesn't really matter tbh

**Review by nanaian — DISMISSED** [review]

File: src/battle/move/auto_jump_765C20.c [hunk 9]
```diff
@@ -1,3 +1,34 @@
 #include "common.h"
 
+#ifdef NON_MATCHING
```
Comment (nanaian) [inline src/battle/move/auto_jump_765C20.c:3]: Would be good to add a comment describing why this is non-matching.

File: src/battle/move/auto_multibounce_76A830.c [hunk 10]
```diff
@@ -1,3 +1,33 @@
 #include "common.h"
 
+#ifdef NON_MATCHING
+ApiStatus func_80238000_6F10E0(ScriptInstance* script, s32 isInitialCall) {
```
Comment (nanaian) [inline src/battle/move/auto_multibounce_76A830.c:4]: Is this the same function as func_802A1000_765C20? If so, leave a comment saying so pls.

File: src/battle/move/auto_smash_763540.c [hunk 11]
```diff
@@ -1,5 +1,31 @@
...
+    }
+
+    return ApiStatus_DONE2;
+}
+
+#ifdef NOT_MATCHING
+ApiStatus func_802A1050_763590(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
+
+    if (is_ability_active(Ability_RIGHT_ON) != FALSE) {
```
Comment (nanaian) [inline src/battle/move/auto_smash_763540.c:19]: Remove `!= FALSE` on both these ifs.

File: src/battle/move/auto_smash_763540.c [hunk 12]
```diff
@@ -1,5 +1,31 @@
...
+        script->varTable[0] = 0;
+        script->varTable[1] = 40;
+    }
+
+    return ApiStatus_DONE2;
+}
+
+#ifdef NOT_MATCHING
+ApiStatus func_802A1050_763590(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
```
Comment (nanaian) [inline src/battle/move/auto_smash_763540.c:17]: `= FALSE`. The assignments inside the ifs can be `TRUE`, too.

File: src/battle/move/auto_smash_763540.c [hunk 13]
```diff
@@ -1,5 +1,31 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "battle/move/auto_smash_763540", func_802A1000_763540);
+ApiStatus func_802A1000_763540(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = *(&gBattleStatus.unk_83);
+    script->varTable[1] = 15;
 
+    if (is_ability_active(Ability_BERSERKER) != FALSE) {
```
Comment (nanaian) [inline src/battle/move/auto_smash_763540.c:7]: Remove `!= FALSE`, just `if (is_ability_active(...))`.

File: src/battle/move/auto_jump_765C20.c (hunk 9 above)
Comment (nanaian) [reply src/battle/move/auto_jump_765C20.c:3]: Same goes for all the nonmatching funcs - even something like "regalloc" or "float issues" would suffice

File: src/battle/move/d_down_pound_74AD90.c [hunk 14]
```diff
@@ -1,7 +1,42 @@
...
+    }
 
-INCLUDE_ASM(s32, "battle/move/d_down_pound_74AD90", func_802A10A4_74AE34);
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A1050_74ADE0(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
+
+    if (is_ability_active(Ability_RIGHT_ON) != FALSE) {
```
Comment (nanaian) [inline src/battle/move/d_down_pound_74AD90.c:18]: Remove `!= FALSE`.

File: src/battle/move/d_down_pound_74AD90.c [hunk 15]
```diff
@@ -1,7 +1,42 @@
...
+        script->varTable[0] = 0;
+        script->varTable[1] = 40;
+    }
 
-INCLUDE_ASM(s32, "battle/move/d_down_pound_74AD90", func_802A10A4_74AE34);
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A1050_74ADE0(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
```
Comment (nanaian) [inline src/battle/move/d_down_pound_74AD90.c:16]: Make these assignments `TRUE` and `FALSE` pls.

File: src/battle/move/hammer_charge_0_74F130.c [hunk 16]
```diff
@@ -1,11 +1,70 @@
...
-INCLUDE_ASM(s32, "battle/move/hammer_charge_0_74F130", func_802A10C8_74F1F8);
+ApiStatus func_802A1050_74F180(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
 
-INCLUDE_ASM(s32, "battle/move/hammer_charge_0_74F130", func_802A11CC_74F2FC);
+    if (is_ability_active(Ability_RIGHT_ON) != FALSE) {
+        script->varTable[0] = 1;
+    }
+
+    if ((gBattleStatus.flags1 & 0x1000) != FALSE) {
```
Comment (nanaian) [inline src/battle/move/hammer_charge_0_74F130.c:22]: Same here for FALSE, TRUE usage in ifs and assignments

File: src/battle/move/hammer_charge_0_74F130.c [hunk 17]
```diff
@@ -1,11 +1,70 @@
...
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A10A4_74F1D4(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    script->varTable[0] = 0;
+
+    if (battleStatus->hammerCharge >= 99) {
+        script->varTable[0] = 1;
```
Comment (nanaian) [inline src/battle/move/hammer_charge_0_74F130.c:34]: Here too

File: src/battle/move/hammer_charge_0_74F130.c [hunk 18]
```diff
@@ -1,11 +1,70 @@
...
+
+    func_800718D0(0, var1, var2, var3, 1.0f, 60);
+    battleStatus->hammerCharge += 1;
+
+    if (battleStatus->hammerCharge >= 100) {
+        (*&battleStatus)->hammerCharge = 99; // TODO: macro?
+    }
+
+    battleStatus->jumpCharge = 0;
+    battleStatus->flags1 = ((battleStatus->flags1 | 0x10000000) & 0xDFFFFFFF);
```
Comment (nanaian) [inline src/battle/move/hammer_charge_0_74F130.c:55]: Change `0xDFFFFFFF` to `~0x20000000`

File: src/battle/move/jump_charge_2_761920.c [hunk 19]
```diff
@@ -1,11 +1,96 @@
...
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A10E4_761A04(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    script->varTable[0] = 0;
+
+    if (battleStatus->jumpCharge >= 99) {
+        script->varTable[0] = 1;
```
Comment (nanaian) [inline src/battle/move/jump_charge_2_761920.c:39]: Same here

File: src/battle/move/jump_charge_2_761920.c [hunk 20]
```diff
@@ -1,11 +1,96 @@
...
+}
+
+ApiStatus func_802A1228_761B48(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 var1 = get_variable(script, *args++);
+    s32 var2 = get_variable(script, *args++);
+    s32 var3 = get_variable(script, *args++);
+    f32 var4 = get_float_variable(script, *args++);
+    s32 var5 = 45;
+    s32 var6 = 30;
```
Comment (nanaian) [inline src/battle/move/jump_charge_2_761920.c:80]: Are these temps required?

File: src/battle/move/mega_quake_756780.c [hunk 21]
```diff
@@ -1,7 +1,29 @@
...
+
+ApiStatus func_802A1050_7567D0(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
+
+    if (is_ability_active(Ability_RIGHT_ON) != FALSE) {
+        script->varTable[0] = 1;
+    }
+
+    if ((gBattleStatus.flags1 & 0x1000) != FALSE) {
+        script->varTable[0] = 1;
```
Comment (nanaian) [inline src/battle/move/mega_quake_756780.c:23]: Same here for booleans

File: src/battle/move/jump_charge_0_74D570.c [hunk 22]
```diff
@@ -1,9 +1,72 @@
...
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A10E4_74D654(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    script->varTable[0] = 0;
+
+    if (battleStatus->jumpCharge >= 99) {
+        script->varTable[0] = 1;
```
Comment (nanaian) [inline src/battle/move/jump_charge_0_74D570.c:36]: Same here

File: src/battle/move/mega_smash_777690.c [hunk 23]
```diff
@@ -1,5 +1,32 @@
...
+#ifdef NOT_MATCHING
+ApiStatus func_802A1050_7776E0(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
+
+    if (is_ability_active(Ability_RIGHT_ON) != FALSE) {
+        script->varTable[0] = 1;
+    }
+
+    if ((gBattleStatus.flags1 & 0x1000) != FALSE) {
+        script->varTable[0] = 1;
```
Comment (nanaian) [inline src/battle/move/mega_smash_777690.c:24]: Same here.....

File: src/battle/move/power_quake_767C90.c [hunk 24]
```diff
@@ -1,5 +1,32 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "battle/move/power_quake_767C90", func_802A1000_767C90);
+ApiStatus func_802A1000_767C90(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = *(&gBattleStatus.unk_83);
+    script->varTable[1] = 15;
 
+    if (is_ability_active(Ability_BERSERKER) != FALSE) {
```
Comment (nanaian) [inline src/battle/move/power_quake_767C90.c:7]: Can remove != False

File: src/battle/move/power_quake_767C90.c [hunk 25]
```diff
@@ -1,5 +1,32 @@
...
+#ifdef NOT_MATCHING
+ApiStatus func_802A1050_767CE0(ScriptInstance* script, s32 isInitialCall) {
+    script->varTable[0] = 0;
+
+    if (is_ability_active(Ability_RIGHT_ON) != FALSE) {
+        script->varTable[0] = 1;
+    }
+
+    if ((gBattleStatus.flags1 & 0x1000) != FALSE) {
+        script->varTable[0] = 1;
```
Comment (nanaian) [inline src/battle/move/power_quake_767C90.c:24]: Same here (I'll stop now)

**Review by nanaian — CHANGES_REQUESTED** [review]
Looks mostly good, besides some stuff relating to booleans.
It would be good for either you (in this PR) or @ethteck (later) to move these functions into a common `.inc.c` for them to be `#include`d like we do the other duplicate funcs.

File: src/battle/move/auto_jump_765C20.c [hunk 26]
```diff
@@ -1,3 +1,34 @@
...
+    f32 posX = partnerActor->currentPos.x;
+    f32 posY = partnerActor->currentPos.y;
+    f32 posZ = partnerActor->currentPos.z;
+    f32 goalX = partnerActor->movePos.goal.x;
+    f32 goalY = partnerActor->movePos.goal.y;
+    f32 goalZ = partnerActor->movePos.goal.z;
+
+    script->varTable[0] = (dist3D(posX, posY, posZ, goalX, goalY, goalZ) * 15.0f) / 100.0f;
+
+    if (script->varTable[0] >= 21) {
```
Comment (pixel-stuck) [inline src/battle/move/auto_jump_765C20.c:17]: > 20 here

File: src/battle/move/auto_multibounce_76A830.c [hunk 27]
```diff
@@ -1,3 +1,33 @@
...
+    f32 posX = partnerActor->currentPos.x;
+    f32 posY = partnerActor->currentPos.y;
+    f32 posZ = partnerActor->currentPos.z;
+    f32 goalX = partnerActor->movePos.goal.x;
+    f32 goalY = partnerActor->movePos.goal.y;
+    f32 goalZ = partnerActor->movePos.goal.z;
+
+    script->varTable[0] = (dist3D(posX, posY, posZ, goalX, goalY, goalZ) * 15.0f) / 100.0f;
+
+    if (script->varTable[0] >= 21) {
```
Comment (pixel-stuck) [inline src/battle/move/auto_multibounce_76A830.c:17]: Here 2

File: src/battle/move/d_down_jump_781920.c [hunk 28]
```diff
@@ -1,5 +1,44 @@
...
+    f32 posY = playerActor->currentPos.y;
+    f32 posZ = playerActor->currentPos.z;
+    f32 goalX = playerActor->movePos.goal.x;
+    f32 goalY = playerActor->movePos.goal.y;
+    f32 goalZ = playerActor->movePos.goal.z;
 
-INCLUDE_ASM(s32, "battle/move/d_down_jump_781920", func_802A10E4_781A04);
+    script->varTable[0] = (dist3D(posX, posY, posZ, goalX, goalY, goalZ) * 15.0f) / 100.0f;
+
+    if (script->varTable[0] >= 21) {
```
Comment (pixel-stuck) [inline src/battle/move/d_down_jump_781920.c:16]: Here 3 (and anywhere else, too)

File: src/battle/move/hammer_charge_0_74F130.c [hunk 29]
```diff
@@ -1,11 +1,70 @@
...
+    Bytecode* args = script->ptrReadPos;
+    BattleStatus* battleStatus = &gBattleStatus;
+    s32 var1 = get_variable(script, *args++);
+    s32 var2 = get_variable(script, *args++);
+    s32 var3 = get_variable(script, *args++);
+
+    func_800718D0(0, var1, var2, var3, 1.0f, 60);
+    battleStatus->hammerCharge += 1;
+
+    if (battleStatus->hammerCharge >= 100) {
```
Comment (pixel-stuck) [inline src/battle/move/hammer_charge_0_74F130.c:50]: similar pattern here, > 99

File: src/battle/move/hammer_charge_1_7595B0.c [hunk 30]
```diff
@@ -1,13 +1,94 @@
...
+    func_800718D0(1, var1, var2, var3, 1.0f, 60);
+    var4 = &D_802A43D0;
+    *var4 = 0;
+    if (battleStatus2->hammerCharge > 0) {
+        *var4 = 1;
+    }
+
+    battleStatus2->hammerCharge += 2;
+
+    if (battleStatus2->hammerCharge >= 100) {
```
Comment (pixel-stuck) [inline src/battle/move/hammer_charge_1_7595B0.c:60]: Here as well (again, anywhere else similar)

File: src/battle/move/jump_73D860.c [hunk 31]
```diff
@@ -1,5 +1,58 @@
...
+}
+
+// NOT WORKING
+#ifdef NON_MATCHING
+ApiStatus func_802A10E4_73D944(ScriptInstance* script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+
+    script->varTable[16] = 1;
+
+    if (playerData->bootsLevel != 1) {
```
Comment (pixel-stuck) [inline src/battle/move/jump_73D860.c:38]: Might be a swit h statement?

File: src/code_1A5830.c [hunk 32]
```diff
@@ -400,7 +402,45 @@ ApiStatus func_8027CC10(ScriptInstance* script, s32 isInitialCall) {
 
 INCLUDE_ASM(s32, "code_1A5830", EnemyDamageTarget);
 
-INCLUDE_ASM(s32, "code_1A5830", EnemyFollowupAfflictTarget);
+//INCLUDE_ASM(s32, "code_1A5830", EnemyFollowupAfflictTarget);
```
Comment (pixel-stuck) [inline src/code_1A5830.c:405]: Remove commented include_asm here

File: src/code_1AF230.c (hunk 8 above)
Comment (pixel-stuck) [reply src/code_1AF230.c:20]: Yeah, imo these should be unk_xx if we don't have a good name

**Review by pixel-stuck — CHANGES_REQUESTED** [review]

**Review by ethteck — CHANGES_REQUESTED** [review]
Awesome work! I think given the duplicate functions, it'll be easier to clean this PR up if we consolidate them into includes. I can work on that or show you how if you'd like.

**Pull request merged**